### ISSUE TITLE
tests/VectorException fails sporadically in CI 

### ISSUE THREAD

**Issue by CodeMonkeyLeet** [Microsoft/openenclave#883]
Specifically in the Kabylake configuration, we've seen the VectorException test fail a couple of times when running CI that recovers on rerun:

```
=== This program is used to test basic vector exception functionalities.
TestCpuidInGlobalConstructors: completed successfully.
TestVectorException: will generate a hardware exception inside enclave!
TestVectorException: hardware exception is handled correctly!
TestGetsecInstruction stack parameters are ok.
Success-Illegal GETSEC raised 2nd chance exception.
The value of cpuidRAX is now: 8
.Success-Unsupported CPUID leaf 00000008 raised 2nd chance exception.
The value of cpuidRAX is now: -2147483648
.Success-Unsupported CPUID leaf 80000000 raised 2nd chance exception.
The value of cpuidRAX is now: 21
.The value of cpuidRAX is now: 591593
.The value of cpuidRAX is now: 1979933441
.The value of cpuidRAX is now: 469762337
.TestSigillHandling: completed successfully.
Test failed: /home/jenkins/workspace/Bors_staging-SAVQQOGG4TMIJW6ZHSGG2MHERTL45R6RNECLBYNXEIQ6O4GKH54Q/tests/VectorException/host/host.c(78): TestSigillHandling cpuid_info[j] == args.cpuid_table[i][j]
```

**Comment by anakrish** [comment]
The reason the test fails is due to the use of cpuid with leaf 0, which returns the current executing processor (initial APIC id) in the higher 8 bits of the ebx register.

When the enclave is created, we compute the cpuids on the host side and cache it in the enclave.
The cached value of cpuid.1 will have the current processor id in ebx.

At a later point, the OS could schedule the enclave to execute in another processor, at which point cpuid.1 when called from the host will have the new processor id where as the cached value in the enclave will have the old processor id.

The test performs such a comparison, and sporadically when the enclave gets scheduled across multiple processors, the test fails.

If is possible to reproduce this by adding a sleep(20) after enclave creation, then changing the affinity of the process via (taskset -p 1,1) so that when the process resumes, it is scheduled to a different processor.

**Comment by gupta-ak** [comment]
Note that intel caches leaves 0, 1, 4, and 7, so they should have the same problem too:
https://github.com/intel/linux-sgx/blob/984c04c696bddaff7fb1621b79760fabcb8c94d7/psw/urts/cpu_features.cpp

**Comment by anitagov** [comment]
Excellent info Anand.  The test should be modified to ignore the APIC id 8-bits in the EBX register or just issue a message if the comparison fails, i.e. no fail the test. 

- Anita

**Comment by anitagov** [comment]
Excellent info Anand.  The test should be modified to ignore the APIC id 8-bits in the EBX register or just issue a message if the comparison fails, i.e. not fail the test.

- Anita

**Comment by anitagov** [comment]
Fixed by #905.
